### PR TITLE
Update GE plugin to 3.11.1

### PR DESCRIPTION
**Pull request by facewindu**
Update Gradle Enterprise Gradle plugin to 3.11

### CODE REVIEW

**Comment by facewindu** [comment]
@bot-gradle test this

**Comment by facewindu** [comment]
@bot-gradle test this

**Review by wolfs — APPROVED** [review]
LGTM!

**Review by wolfs — APPROVED** [review]
LGTM!

File: settings.gradle.kts [hunk 1]
```diff
@@ -17,7 +17,7 @@ pluginManagement {
 }
 
 plugins {
-    id("com.gradle.enterprise").version("3.10.3")
+    id("com.gradle.enterprise").version("3.11")
     id("io.github.gradle.gradle-enterprise-conventions-plugin").version("0.7.6")
     id("gradlebuild.base.allprojects")
     id("com.gradle.enterprise.test-distribution").version("2.3.5") // Sync with `build-logic/build-platform/build.gradle.kts`
```
Comment (wolfs) [inline settings.gradle.kts:23]: Actually, this application needs to go away, and the other build file(s) probably need to be changed now.

**Review by wolfs — CHANGES_REQUESTED** [review]

**Comment by facewindu** [comment]
@bot-gradle test this

File: build-logic/settings.gradle.kts [hunk 2]
```diff
@@ -28,7 +28,6 @@ dependencyResolutionManagement {
             content {
                 val rcAndMilestonesPattern = "\\d{1,2}?\\.\\d{1,2}?(\\.\\d{1,2}?)?-((rc-\\d{1,2}?)|(milestone-\\d{1,2}?))"
                 includeVersionByRegex("com.gradle", "gradle-enterprise-gradle-plugin", rcAndMilestonesPattern)
-                includeVersionByRegex("com.gradle.enterprise", "test-distribution-gradle-plugin", rcAndMilestonesPattern)
```
Comment (wolfs) [inline build-logic/settings.gradle.kts:31]: Is that all in one JAR now?

**Review by wolfs — APPROVED** [review]
LGTM!

File: build-logic/build-platform/build.gradle.kts [hunk 3]
```diff
@@ -18,7 +18,6 @@ val kotlinVersion = providers.gradleProperty("buildKotlinVersion")
 dependencies {
     constraints {
         api("com.gradle:gradle-enterprise-gradle-plugin:3.11")
-        api("com.gradle.enterprise:test-distribution-gradle-plugin:2.3.5") // Sync with `settings.gradle.kts`
```
Comment (marcphilipp) [inline build-logic/build-platform/build.gradle.kts:21]: I think the comment should be preserved and put on the line above.

File: settings.gradle.kts [hunk 4]
```diff
@@ -20,7 +18,6 @@ plugins {
     id("com.gradle.enterprise").version("3.11")
     id("io.github.gradle.gradle-enterprise-conventions-plugin").version("0.7.6")
     id("gradlebuild.base.allprojects")
-    id("com.gradle.enterprise.test-distribution").version("2.3.5") // Sync with `build-logic/build-platform/build.gradle.kts`
```
Comment (marcphilipp) [inline settings.gradle.kts:23]: Ditto.

File: build-logic/settings.gradle.kts (hunk 2 above)
Comment (facewindu) [reply build-logic/settings.gradle.kts:31]: yes

File: build-logic/build-platform/build.gradle.kts (hunk 3 above)
Comment (facewindu) [reply build-logic/build-platform/build.gradle.kts:21]: done

File: settings.gradle.kts (hunk 4 above)
Comment (facewindu) [reply settings.gradle.kts:23]: done

**Comment by marcphilipp** [comment]
@bot-gradle test this

**Comment by marcphilipp** [comment]
@bot-gradle test this

**Comment by big-guy** [comment]
@bot-gradle test this

**Comment by big-guy** [comment]
Thanks @wolfs, if this unblocks you, let's merge it.  We'll follow up with #13955

**Comment by wolfs** [comment]
@bot-gradle test and merge

**Comment by wolfs** [comment]
@bot-gradle test and merge

**Comment by wolfs** [comment]
@bot-gradle test and merge

**Pull request merged**